### PR TITLE
[Remote Config] Add exponential backoff to HTTP client

### PR DESCRIPTION
**Pull request by ameske**
### What does this PR do?

Adds exponential backoff to the remote config HTTP client to
prevent agents from overwhelming the remote config edge during service
delivery issues.

### Motivation

The Remote Configuration backend will be responsible for serving
a large number of agents in the field. At the moment these agents
all poll Remote Config at a default polling interval of 1 minute.
Given the expected scope of the deployments, when the Remote Config
backend is unstable, it would be beneficial for agents to quickly
back their polling rates off until we recover.

### Additional Notes

This work leverages the existing `backoff` package already contained
within the agent codebase.

### Possible Drawbacks / Trade-offs

If errors occur, agents could have a delay receiving updated config
once the service is stable. However, to counteract this the maximum
backoff interval is constrained to 5 minutes, and the recovery rate
is quick.

### Describe how to test/QA your changes

Unit tests are provided for how the backoff interval is calculated.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by jtappa — COMMENTED** [review]
👍🏻 for docs/release notes.

**Review by jtappa — APPROVED** [review]

**Review by mellon85 — APPROVED** [review]

File: pkg/config/remote/service/service_test.go [hunk 1]
```diff
@@ -84,6 +86,93 @@ func newTestService(t *testing.T, api *mockAPI, uptane *mockUptane, clock clock.
...
+	api.On("Fetch", mock.Anything, &pbgo.LatestConfigsRequest{
+		Hostname:                     service.hostname,
+		AgentVersion:                 version.AgentVersion,
+		CurrentConfigSnapshotVersion: 0,
+		CurrentConfigRootVersion:     0,
+		CurrentDirectorRootVersion:   0,
+		Products:                     []string{},
+		NewProducts:                  []string{},
+	}).Return(lastConfigResponse, nil)
+	service.api = api
```
Comment (hush-hush) [inline pkg/config/remote/service/service_test.go:156]: This seems like a second test case. Could we split this test into more "unit" like one ?

File: releasenotes/notes/remote-config-http-client-backoff-d9acbc7bc24b4ca7.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    The Remote Configuration client will do exponential backoff
+    for subsequent requests when errors are encountered to preserve
+    service stability.
```
Comment (hush-hush) [inline releasenotes/notes/remote-config-http-client-backoff-d9acbc7bc24b4ca7.yaml:13]: Is remote configuration public ? I could not find anything in our release note and the configuration isn't in the config template (which make me this it's still internal/beta)

Release notes are used to generate the CHANGELOG which is meant for users not dev. We add item to it only if we want any user/customer to use the new feature. I guess this PR should not have a release note. You can use the `changelog/no-changelog` Github label to pass the CI.

**Review by hush-hush — CHANGES_REQUESTED** [review]

File: pkg/config/remote/service/service_test.go (hunk 1 above)
Comment (ameske) [reply pkg/config/remote/service/service_test.go:156]: Yep, can do!

File: releasenotes/notes/remote-config-http-client-backoff-d9acbc7bc24b4ca7.yaml (hunk 2 above)
Comment (ameske) [reply releasenotes/notes/remote-config-http-client-backoff-d9acbc7bc24b4ca7.yaml:13]: It's still only internal.

**Review by hush-hush — APPROVED** [review]

**Pull request merged**